### PR TITLE
Upgrade iOS Build to Qt6

### PR DESCRIPTION
**Pull request by ann0see**
**Short description of changes**
Upgrades the iOS build to Qt6. There are still some UI issues (e.g. the mixer has some overflows, see https://github.com/jamulussoftware/jamulus/issues/3406)

CHANGELOG: Upgrade iOS Build to Qt 6
**Context: Fixes an issue?**

Related to: https://github.com/jamulussoftware/jamulus/issues/3406

**Does this change need documentation? What needs to be documented and how?**

Yes. This will make the UX worse due to UI bugs introduced with Qt6.

**Status of this Pull Request**

Testing on an actual device. Will do that soon.

**What is missing until this pull request can be merged?**
Testing.

## Checklist

-  [x] I've verified that this Pull Request follows the [general code principles](https://github.com/jamulussoftware/jamulus/blob/main/CONTRIBUTING.md#jamulus-projectsource-code-general-principles)
-  [x] I tested my code and it does what I want
-  [x] My code follows the [style guide](https://github.com/jamulussoftware/jamulus/blob/main/CONTRIBUTING.md#source-code-consistency) 
-  [x] I waited some time after this Pull Request was opened and all GitHub checks completed without errors. 
-  [x] I've filled all the content above

### CODE REVIEW

File: .github/autobuild/ios.sh [hunk 1]
```diff
@@ -45,7 +45,15 @@ setup() {
...
-        python3 -m pip install "aqtinstall==${AQTINSTALL_VERSION}"
+        # We may need to create the Qt installation directory and chown it to the runner user to fix permissions
+        sudo mkdir -p "${QT_DIR}"
+        sudo chown "$(whoami)" "${QT_DIR}"
+        # Create and enter virtual environment
+        python3 -m venv venv
+        # Must hide directory as it just gets created during execution of the previous command and cannot be found by shellcheck
+        # shellcheck source=/dev/null
+        source venv/bin/activate
+        pip install "aqtinstall==${AQTINSTALL_VERSION}"
```
Comment (ann0see) [inline .github/autobuild/ios.sh:56]: Changed to be in line with macOS

File: .github/autobuild/ios.sh [hunk 2]
```diff
@@ -59,13 +67,12 @@ setup() {
             # Starting with Qt6, ios' qtbase install does no longer include a real qmake binary.
             # Instead, it is a script which invokes the mac desktop qmake.
             # As of aqtinstall 2.1.0 / 04/2022, desktop qtbase has to be installed manually:
-            python3 -m aqt install-qt --outputdir "${QT_DIR}" mac desktop "${QT_VERSION}" --archives qtbase
+            python3 -m aqt install-qt --outputdir "${QT_DIR}" mac desktop "${QT_VERSION}" --archives qtbase qttools
```
Comment (ann0see) [inline .github/autobuild/ios.sh:70]: Necessary to have lrelease in the correct spot. Otherwise the build fails.

**Comment by ann0see** [comment]
Tested, and it works as expected (with the known UI bugs)

**Comment by ann0see** [comment]
Fixes: https://github.com/jamulussoftware/jamulus/issues/3433

File: .github/workflows/autobuild.yml [hunk 3]
```diff
@@ -250,15 +250,13 @@ jobs:
...
-            building_on_os:         macos-12
-            base_command:           QT_VERSION=5.15.2 ./.github/autobuild/ios.sh
+            building_on_os:         macos-14
+            base_command:           QT_VERSION=6.7.3 ./.github/autobuild/ios.sh
             # Build failed with CodeQL enabled when last tested 03/2022 (#2490).
             # There are no hints that iOS is supposed to be supported by CodeQL.
             # Therefore, disable it:
             run_codeql:             false
-            # Unfortunately, more modern Xcode versions no longer seem to support
-            # Qt 5.15.2. Therefore upgrading to Qt6 is needed in the medium term (#2711)
```
Comment (pljones) [inline .github/workflows/autobuild.yml:260]: So does this "Fixes: #2711", too?

**Review by pljones — APPROVED** [review]
Approved on the grounds the iOS build completed.

**Pull request merged**